### DISCUSSION TITLE
Caching and storing media from feeds forever

### DISCUSSION THREAD

**Discussion by squromiv** [FreshRSS/FreshRSS#4486] [General]
Often I use FreshRSS feeds as archives of data from sources, which usually become unavailable. I very miss the feature of caching and storing media (pictures, video, audio) from feeds forever (outside of database). Is it possible somehow to achieve such a purpose with FreshRSS or third-party applications? Ideally:
 - "Store media"checkbox in properties of feed
 - deleting a feed results to deleting of media files

Is realization of such a function looks unreal?

**Comment by squromiv** [comment]
I know, that there is the [issue](https://github.com/FreshRSS/FreshRSS/issues/3798) about images. But I am interested in saving all files placed in feeds.

**Reply by Frenzie** [reply]
You'd have to write something not dissimilar in some ways to how the ImageProxy extension works but significantly more involved in order to store the binary data somewhere. It's not undoable but I don't see someone writing it in an afternoon.

**Comment by Alkarex** [comment]
See https://github.com/FreshRSS/Extensions/pull/139

**Reply by squromiv** [reply]
> See [FreshRSS/Extensions#139](https://github.com/FreshRSS/Extensions/pull/139)

I do not understand how to use this extension in order to store images in local instance (Windows Apache Web Server). Could you try to describe step by step?  Should I run the other one proxy or make any config for my local server?

**Reply by Alkarex** [reply]
I have not tried that extension myself, but you might be able to ask the author for additional explanations if needed https://github.com/FreshRSS/Extensions/pull/139
